라이더 출근/퇴근 토글 기능

handsone-u opened the pull request:
> 출근 상태인 라이더를 Set 자료구조로 Redis에 보관하도록 구현했습니다.

- PATCH: `/v0/me/duty`
- Request: `record ToggleDutyRequest(Boolean onDuty)`

하나의 API로 onDuty 값에 따라 출근 상태를 토글할 수 있도록 구현했습니다.

'**on_duty**' key 값으로 출근 상태인 라이더의 Id를 value로 Set<Long>에 넣습니다.
근데 JDK 직렬화가 key, value 모두 적용되기에 redis-cli에서 '**on_duty**' 값으로 조회할 수는 없습니다.

handsone-u committed:
chore: 패키지 수정

handsone-u committed:
[#6] add: mapstruct 의존성 추가

handsone-u committed:
[#6] add: data-redis 의존성 추가

handsone-u committed:
[#6] add: redis 설정 추가

handsone-u committed:
[#6] del: usecase 층 제거

handsone-u committed:
[#6] feat: 라이더 출근/퇴근 토글 API

JinTaekLim commented on build.gradle:
의존성별 버전 명시가 필요해보입니다.

JinTaekLim commented on src/main/java/com/baedal/rider/adapter/cache/repository/RiderRosterAdapter.java:
```diff
@@ -0,0 +1,49 @@
+@Repository
+@RequiredArgsConstructor
+@Slf4j
+public class RiderRosterAdapter implements RiderRosterPort {
```
Adapter 에서 redisTemplate 을 바로 사용하기 보단, 다른 패키지 구조의 Jpa interface 와 같이 활용하는 편이 좋을 것같습니다.

handsone-u replied:
redisTemplate를 사용하는 모듈을 만든다면 어떤 이점이 있나요?

JinTaekLim replied:
Redis 에서 다른 저장소로 변경하게 되더라도 Adapter 의 변경 없이 변경할 수 있다는 이점, 이후 다른 로직이 포함되었을 때 가독성 등의 이점이 있을 같아요.

Adatper 내부 코드를 변경하는 방안도 있겠지만, 전자 쪽이 조금 더 유지보수에 용이할 것같아요.

JinTaekLim commented on src/main/java/com/baedal/rider/adapter/presentation/controller/RiderController.java:
```diff
@@ -2,34 +2,50 @@
+      service.makeOnDuty(riderId);
+    } else {
+      service.makeOffDuty(riderId);
+    }
```
위와 같은 로직은 service 단에 작성하는게 적합해 보입니다.

JinTaekLim commented on src/main/java/com/baedal/rider/adapter/presentation/controller/RiderController.java:
```diff
@@ -2,34 +2,50 @@
+  }
+
+  @PreAuthorize("hasRole('RIDER')")
+  @PatchMapping("/v0/me/duty")
```
공통 매핑 주소는 어노테이션으로 컨트롤러 상단에 위치하는게 좋아보입니다

handsone-u replied:
/v0 는 각 API 마다 버전 명시하는거라 /v1 같이 다른 버전 나오면 공통 매핑 어노테이션을 쓰지 못하게 될것 같아요. 그래서 /v0는 각각 명시하긴 했는데 /v1 필요하면 다른 컨트롤러 만드는게 나을것 같기도 하네요. 반영하겠습니다.

handsone-u committed:
[#6] add: 라이브러리 버전 명시

handsone-u committed:
[#6] refactor: 라이터 출/퇴근 토글 API

- 제어 변수를 응용 계층에서 처리하도록 변경

JinTaekLim reviewed (approved):
Controller 공통 매핑 주소만 조치해주신 후 merge 하시면 될 것같습니다!

handsone-u committed:
[#6] fix: Generic raw 타입 경고

handsone-u committed:
[#6] add: RiderRosterAdapter 컴포넌트 어노테이션

- Repository -> Component
- @Repository는 "DB 예외를 스프링이 변환(Exception Translation)해준다"는 역할
- RiderRosterAdapter는 Spring Data JPA나 MyBatis 같은 DB 컴포넌트가 아니라 Redis를 직접 다루는 것

handsone-u committed:
[#6] refactor: redis store 분리

- cache 기술과 관련된 로직을 store가 관리

handsone-u committed:
[#6] add: 매핑 url 수정

- 도메인 이름을 붙이도록
